### PR TITLE
Allow admin users to modify hostnames in web UI

### PR DESCRIPTION
**Pull request by jonathon-k**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Provides a box for administrative users (user level 10) to rename hosts from within the web UI on the edit device page

### CODE REVIEW

File: html/pages/device/edit/device.inc.php [hunk 1]
```diff
@@ -40,6 +40,25 @@
         } else {
             $update_message = "Device record update error.";
         }
+
+        if ($_POST['hostname'] !== '' && $_POST['hostname'] !== $device['hostname']) {
+            if ($_SESSION['userlevel'] === 10) {
```
Comment (laf) [inline html/pages/device/edit/device.inc.php]: We have functions to check admin levels, something like is_admin(), I can't remember 100%

**Comment by laf** [comment]
Looks good to me.

Can't help but feel we need something to indicate users can change the hostname though.

**Comment by jonathon-k** [comment]
You mean before they try to change it?

**Comment by jonathon-k** [comment]
Except it is beginning to look like only admin users have can access the edit page...

**Comment by laf** [comment]
No need for a new user level, I just mean make it obvious what changing that does, people might think it's just an alias and change it.

**Comment by laf** [comment]
Something like this:

image

Clicking just toggles disabled/enabled to allow editing.

**Comment by laf** [comment]
Diff if you want it:

https://gist.github.com/laf/d4b66af4e1da113a768991f22c3bec4f

**Pull request merged**